e2e/cypress/integration/search_filter/on_spec.js

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @search_date_filter
+
+import {getAdminAccount} from '../../support/env';
+
+import {
+    getMsAndQueryForDate,
+    getTestMessages,
+    searchAndValidate,
+    setupTestData,
+} from './helpers';
+
+describe('SF15699 Search Date Filter - on', () => {
+    const testData = getTestMessages();
+    const {
+        commonText,
+        firstDateEarly,
+        secondDateEarly,
+        secondMessage,
+        secondOffTopicMessage,
+    } = testData;
+    const admin = getAdminAccount();
+    let anotherAdmin;
+
+    before(() => {
+        cy.apiInitSetup({userPrefix: 'other-admin'}).then(({team, user}) => {
+            anotherAdmin = user;
+
+            setupTestData(testData, {team, admin, anotherAdmin});
+        });
+    });
+
+    it('omits results before and after target date', () => {
+        searchAndValidate(`on:${secondDateEarly.query} ${commonText}`, [secondOffTopicMessage, secondMessage]);
+    });
+
+    it('takes precedence over "after:" and "before:"', () => {
+        searchAndValidate(`before:${Cypress.moment().format('YYYY-MM-DD')} on:${secondDateEarly.query} ${commonText}`, [secondOffTopicMessage, secondMessage]);
+    });
+
+    it('takes precedence over "after:"', () => {
+        searchAndValidate(`after:${firstDateEarly.query} on:${secondDateEarly.query} ${commonText}`, [secondOffTopicMessage, secondMessage]);
+    });
+
+    it('can be used in conjunction with "in:"', () => {
+        searchAndValidate(`on:${secondDateEarly.query} in:town-square ${commonText}`, [secondMessage]);
+    });
+
+    it('can be used in conjunction with "from:"', () => {
+        searchAndValidate(`on:${secondDateEarly.query} from:${anotherAdmin.username} ${commonText}`, [secondOffTopicMessage]);
+    });
+
+    it('works from 12:00am to 11:59pm', () => {
+        // create posts on a day att 11:59 the previous day, 12:00am the main day, 11:59pm the main day, and 12:00 the next day
```

Comment:
Typo: `at` instead of `att`
```suggestion
        // create posts on a day at 11:59 the previous day, 12:00am the main day, 11:59pm the main day, and 12:00 the next day
```

Reply:
Good catch, done and thanks.